engine/table/src/main/java/io/deephaven/engine/table/impl/by/DoubleChunkedVarOperator.java

```diff
@@ -97,8 +97,26 @@ private boolean addChunk(DoubleChunk<? extends Values> values, long destination,
             if (forceNanResult || nonNullCount <= 1) {
                 resultColumn.set(destination, Double.NaN);
             } else {
-                final double variance = (newSum2 - newSum * newSum / nonNullCount) / (nonNullCount - 1);
-                resultColumn.set(destination, std ? Math.sqrt(variance) : variance);
+                // If the sum has reach +/-Infinity, we are stuck with NaN forever.
+                if (Double.isInfinite(newSum) || Double.isInfinite(newSum2)) {
```

Comment:
Yes, this snippet still works (producing NaN when `Inf` is in the tail, a normal output otherwise).

The code in question only handles covers the case where we have reached `Inf` by summing `normal` values.  At that point the agg code would be subtracting or adding `normals` from `inf` with each new/deleted/modified value and we can short-circuit some the following code by assigning `NaN` immediately.